pkg/storageincentives/redistributionstate.go

```diff
@@ -50,164 +50,125 @@ type Status struct {
 	Fees            *big.Int
 }
 
-func NewRedistributionState(logger log.Logger, stateStore storage.StateStorer, erc20Service erc20.Service, contract transaction.Service) *RedistributionState {
-	return &RedistributionState{
+func NewRedistributionState(logger log.Logger, ethAddress common.Address, stateStore storage.StateStorer, erc20Service erc20.Service, contract transaction.Service) (*RedistributionState, error) {
+	s := &RedistributionState{
+		ethAddress:     ethAddress,
 		stateStore:     stateStore,
 		erc20Service:   erc20Service,
 		logger:         logger.WithName(loggerNameNode).Register(),
 		currentBalance: big.NewInt(0),
 		txService:      contract,
-		status: Status{
+		status: &Status{
 			Reward: big.NewInt(0),
 			Fees:   big.NewInt(0),
 		},
-		saveStatusC: make(chan Status, 8),
-	}
-}
-
-func (n *Status) clone() Status {
-	return Status{
-		Phase:           n.Phase,
-		IsFrozen:        n.IsFrozen,
-		Round:           n.Round,
-		LastWonRound:    n.LastWonRound,
-		LastPlayedRound: n.LastPlayedRound,
-		LastFrozenRound: n.LastFrozenRound,
-		Block:           n.Block,
-		Reward:          big.NewInt(0).Set(n.Reward),
-		Fees:            big.NewInt(0).Set(n.Fees),
 	}
-}
-func (n *RedistributionState) Stop() {
-	close(n.saveStatusC)
-}
-
-func (n *RedistributionState) Start() {
-	go n.saveStateHandler()
-}
-
-func (n *RedistributionState) saveStateHandler() {
-	var status *Status // when nil latests is saved, when not nil status is dirty
-
-	saveStatus := func() {
-		if status == nil {
-			return
-		}
-
-		err := n.stateStore.Put(redistributionStatusKey, *status)
-		if err != nil {
-			n.logger.Error(err, "error saving node status")
-		} else {
-			status = nil
-		}
-	}
-
-	// save status is called in deffer to ensure that
-	// state is stored before handler exists
-	defer saveStatus()
 
-	saveTicker := time.NewTicker(saveStatusInterval)
-	defer saveTicker.Stop()
-
-	for {
-		select {
-		case newStatus, ok := <-n.saveStatusC:
-			if !ok {
-				return
+	status, err := s.Status()
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			s.status = &Status{
+				Reward: big.NewInt(0),
+				Fees:   big.NewInt(0),
 			}
-
-			status = &newStatus
-
-		case <-saveTicker.C:
-			saveStatus()
+			return s, nil
 		}
+		return nil, err
 	}
-}
 
-func (n *RedistributionState) SetCurrentEvent(p PhaseType, r uint64, b uint64) {
-	n.mtx.Lock()
-	defer n.mtx.Unlock()
-	n.status.Phase = p
-	n.status.Round = r
-	n.status.Block = b
+	s.status = status
+	return s, nil
+}
 
-	n.saveStatusC <- n.status.clone()
+func (r *RedistributionState) SetCurrentEvent(p PhaseType, round uint64, b uint64) {
```

Comment:
Could we have the full name for all the parameters or just the initials of properties for consistency purposes? Same for other function params.

Like `SetCurrentEvent(phase PhaseType, round uint64, block uint64)`